packages/components/src/components/hds/app-side-nav/list/index.hbs

```diff
@@ -3,7 +3,7 @@
   SPDX-License-Identifier: MPL-2.0
 }}
 
-<nav class="hds-app-side-nav__list-wrapper" ...attributes>
+<nav class="hds-app-side-nav__list-wrapper" aria-labelledby="hds-app-side-nav-header" ...attributes>
   {{yield (hash ExtraBefore=(component "hds/yield"))}}
   <ul class="hds-app-side-nav__list" role="list">
```

Comment:
This is where you'd add an aria-labelledby attribute, something like:

```suggestion
  <ul class="hds-app-side-nav__list" role="list" aria-labelledby="listTitle-id">
```

And then in the `app-side-nav/list-title`, add a `guidFor` that produces a unique `id` for the title. 

If we associate these correctly, the user would not have to change anything on their end.

Reply:
Thanks I pushed up changes to add an id to each Title and generate a list of TitleIds that gets inserted in the aria-labelledby of the ul.

Let me know if I should remove the h3s I added earlier for the Titles.

Reply:
@MelSumner @shleewhite When you have time could you please review the latest changes I made from @MelSumner's feedback above?